docker/Dockerfile

```diff
@@ -0,0 +1,21 @@
+
+FROM astral/uv:bookworm-slim
+
+ARG CARROT_VERSION
+ENV CARROT_VERSION=${CARROT_VERSION}
+
+RUN uvx carrot-transform@${CARROT_VERSION} -v
```

Comment:
From my understanding, this is just going to go and get the package from pypi, and not from the source code in the repository.

This isn't something we want - the dockerfile should build from this source code in the repo, not just run it in a container. 

Installing the tool from pypi is a too extreme view of keeping it simple!